src/components/MultiHopTrade/hooks/useAllowanceApproval/hooks/useExecuteAllowanceApproval.tsx

```diff
@@ -18,12 +19,15 @@ export const useExecuteAllowanceApproval = (
   buildCustomTxInput?: evm.BuildCustomTxInput,
 ) => {
   const [txId, setTxId] = useState<string | undefined>()
+  const [txStatus, setTxStatus] = useState<TxStatus>(TxStatus.Unknown)
```

Comment:
Not strictly related to this PR and obviously doesn't need to be tackled for now but do we want to expose loading states as well vs. relying on undefined checks? Perhaps some error state being exposed as well.

Which makes me wonder whether or not wagmi has a hook doing precisely that for us ?👀

Reply:
interesting, will take a look as this will smooth things out nicely